frontend/app/utils/application-code-utils.ts

```diff
@@ -10,3 +10,14 @@ export function isValidApplicationCode(applicationCode: string): boolean {
   const applicationCodeRegex = /^(?:\d{13}|\d{11}|\d{6})$/;
   return applicationCodeRegex.test(applicationCode);
 }
+
+/**
+ *
+ * @param applicationCode the 13 digit application code returned from PP after an applicant submits their application
+ * @returns submission code in the format of XXX XXX XXX XXXX
+ */
+export function formatSubmissionApplicationCode(applicationCode: string): string {
+  const strippedCode = applicationCode.replace(/ /g, '');
+  if (!/^\d{13}$/.test(strippedCode)) throw new Error('submission application code must be 13 digits');
```

Comment:
Since we don't maintain the Power Platform API and don't have control over how many digits is returned to us, should we be throwing here? If submission to Power Platform is successful, and the returned confirmation code is not 13 digits, maybe we should return an unformulated confirmation code instead.

Reply:
yeah, that's reasonable.